spec/models/settings/base_spec.rb

```diff
@@ -0,0 +1,169 @@
+require "rails_helper"
+
+RSpec.describe Settings::Base, type: :model do
+  with_model :TestSetting, superclass: described_class do
+    table do |t|
+      t.string :var, null: false
+      t.text :value, null: true
+      t.timestamps
+    end
+
+    model do
```

Comment:
Here we put the actual content of our model, e.g. what we would otherwise put in `app/models/test_setting.rb`.